### ISSUE TITLE
Hidden Locations

### ISSUE THREAD

**Issue by mdovey** [bic-org-uk/bic-lcf#240]
@dkarvel raised a number of issues as regard location in issue #132 

Most of these issues were address in 1.2.0 with the exception of hidden locations

The proposal was to include additional properties for Location if the Location is hidden i.e. from searching manifestations

**Comment by mdovey** [comment]
The general consensus is that Location does not need any hidden properties. However, an LCF implementation may return different lists of Locations depending on the authenticated user and\or terminal.

This would generally apply to any LCF operation - the list returned would be dependent on the authenticated user\terminal an different users\terminals might be presented with different information.

What has not been established is whether this is sufficient for @dkarvel's original use case

**Comment by franciscave** [comment]
I've re-read the original issue posted by @dkarvel and the ensuing exchanges. I think that the requirement can be re-stated as follows:

A library may wish to hide a Location from "end users" (a category of terminal user) but not from "library staff" (another category of terminal user). It was therefore proposed that there should be a property of a Location that indicates whether it should be hidden from "end users" or not.

The response to this request so far has been to suggest that, while an LMS might need to specify that a Location is to be hidden from end users, this information would not need to be communicated using LCF. When an LMS receives a request for Location information, the LMS would check the terminal user's identity (which would be known if the terminal application supports patron authentication) before deciding whether or not to service the request (i.e. HTTP 200 or HTTP 403). By this argument, LCF would never need to communicate whether or not a Location is hidden: a request for Location information would be either accepted (200) or refused (403).

However, this doesn't fully answer the original requirement, which is that a library staff member, using the same terminal application, should be able to retrieve the Location information, and part of that information would be to indicate that the Location information is hidden from end users. A library staff member should also be able to use LCF to update the Location information to either hide or un-hide it from end users.

I think that this does highlight an assumption that we (or at least I) may have been making: that library staff with direct access to the LMS don't need to use LCF to perform library management functions on the LMS. Using LCF for that kind of task is not a use case that we have been considering, as I see it.

Why are Locations any different from, say, Patrons? In general, all Patron information would be hidden from end users, with the exception of the end user's own Patron information. I think that the difference is that, while there is a fairly simple, general rule about hiding Patron information from end users not entitled to retrieve it, whether or not to hide Location information might need to be determined on a case-by-case basis, and hence the possible need for a binary hidden-or-not flag on each Location.

In conclusion, I think that there is a clear need to be able to specify _on an LMS_ that a Location is to be hidden from end users, but I don't think we have a clear requirement to be able to communicate that using LCF, given that LCF does not at present cover the use case of terminal applications being used for general library management tasks (which would encompass far more than this one particular requirement).

**Comment by mdovey** [comment]
Agreed to close by Review Group as we believe the issue to have been addressed.
